### PR TITLE
Additions/52 lazy load component

### PR DESCRIPTION
**Pull request by LMFinney**
## Checklist for challenge submission

- [ ] Start your PR message with Answer:${challenge_number}

## Warning:

- If you want feedback or review, you must support the project on GitHub:
  - [$5 per review](https://github.com/sponsors/tomalaforge)
  - [$25 for lifetime reviews](https://github.com/sponsors/tomalaforge)
  - Create one of more challenges
  - Have many valuable contributions to the project

Alternatively, you can still submit your PR to join the list of answered challenges or to be reviewed by a community member. 🔥

### CODE REVIEW

File: docs/src/content/docs/challenges/angular/52-lazy-load-component.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+title: 🟢 Lazy Load a Component
+description: Challenge 52 is about understanding how to lazy load a component in Angular.
+author: lance-finney
+contributors:
+  - tomalaforge
```
Comment (tomalaforge) [inline docs/src/content/docs/challenges/angular/52-lazy-load-component.md:6]: I don't need to be quoted here, I didn't do anything 😅

Comment (LMFinney) [reply docs/src/content/docs/challenges/angular/52-lazy-load-component.md:6]: I think that was in the boilerplate. I'll update it.

File: docs/src/content/docs/challenges/angular/52-lazy-load-component.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+---
+title: 🟢 Lazy Load a Component
+description: Challenge 52 is about understanding how to lazy load a component in Angular.
+author: lance-finney
```
[the remark this answers is not in the document]
Comment (LMFinney) [reply docs/src/content/docs/challenges/angular/52-lazy-load-component.md:4]: I'll do that. Thanks

File: docs/src/content/docs/challenges/angular/52-lazy-load-component.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+You should be able to remove the `topLoaded` signal when you are done.
+
+</details>
+
+<details>
+  <summary>Hint 2</summary>
+
+The new Angular feature will hide the `TopComponent` from view, but it will still be loaded in the initial bundle unless you change how both `AppComponent` and `TopComponent` are defined in their decorators. This challenge start with the old `NgModule`-based architecture, but you will need to change it to use the new feature.
+
+</details>
```
Comment (tomalaforge) [inline docs/src/content/docs/challenges/angular/52-lazy-load-component.md:43]: nice challenge,

you can even add a hint 3 saying that we need to use the defer block with a link to the documeantion.

File: apps/angular/52-lazy-load-component/src/app/app.module.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import { AppComponent } from './app.component';
+import { PlaceholderComponent } from './placeholder.component';
+import { TopComponent } from './top.component';
+
+@NgModule({
+  declarations: [AppComponent, PlaceholderComponent, TopComponent],
+  imports: [BrowserModule],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
```
[the remark this answers is not in the document]
Comment (LMFinney) [reply apps/angular/52-lazy-load-component/src/app/app.module.ts:12]: I wanted to drive home that standalone was necessary for this to work fully. It's also a _really_ simple challenge without that step.

If you'd prefer the starting point be with standalone, I can make that change.

**Comment by jdegand** [comment]
Need to update `Go to lastest challenge` link in the actions of `docs/index.mdx`.

Maybe, shorten the challenge title to `lazy load`.

I'd leave the module.

**Comment by LMFinney** [comment]
> Need to update `Go to lastest challenge` link in the actions of `docs/index.mdx`.

Good catch. Thanks.

> Maybe, shorten the challenge title to `lazy load`.

I'm not comfortable with that, because I think most people would think it means router lazy loading. A shorter name would be `deferrable view`, but I think that would give things away.

> I'd leave the module.

I'm fine either way on the NgModules. Hmmm...

**Comment by jdegand** [comment]
I see your point.  `Lazy component` could work.

**Review by tomalaforge — APPROVED** [review]

**Pull request merged**